x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/conditional/Case.java

```diff
@@ -261,6 +267,11 @@ public Expression partiallyFold() {
                 continue;
             }
             modified = true;
+            Object o = condition.condition.fold();
+            if (o instanceof List) {
+                // multivalued field folds to null which folds to false
+                continue;
+            }
```

Comment:
Maybe we should use a helper method (or two) on `Condition` to tell us whether the condition is always true/false? That could be reused in `Case.foldable` and would avoid the foldability logic to drift apart.